sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/utils/DockerUtil.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.trace.aws.resource.utils;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+public class DockerUtil {
+
+  private static final Logger logger = Logger.getLogger(DockerUtil.class.getName());
+  private static final int CONTAINER_ID_LENGTH = 64;
+  private static final String DEFAULT_CGROUP_PATH = "/proc/self/cgroup";
+  private static final DockerUtil INSTANCE = new DockerUtil();
+
+  private DockerUtil() {}
+
+  public static DockerUtil getInstance() {
+    return INSTANCE;
+  }
+
+  /**
+   * Get docker container id from local cgroup file.
+   *
+   * @return docker container ID
+   */
+  @Nullable
+  @SuppressWarnings("DefaultCharset")
+  public String getContainerId() {
+    try (BufferedReader br = new BufferedReader(new FileReader(DEFAULT_CGROUP_PATH))) {
+      String line;
+      while ((line = br.readLine()) != null) {
+        if (line.length() > CONTAINER_ID_LENGTH) {
```

Comment:
Can you find some documentation to support this logic? I guess it probably works in practice, but seems fragile to me. It would be less fragile if we at least checked for `/docker/`. Here's an example with my current docker for mac

```
-> % docker run -it --rm bash
Unable to find image 'bash:latest' locally
latest: Pulling from library/bash
df20fa9351a1: Pull complete
f5b2321f64f1: Pull complete
a82fbae36939: Pull complete
Digest: sha256:376bd611ec5dfc1248a2d21d921bd62d0b858344ce1516cd133249c0a814387d
Status: Downloaded newer image for bash:latest
bash-5.0#
bash-5.0#
bash-5.0# cat /proc/self/cgroup
14:name=systemd:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
13:rdma:/
12:pids:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
11:hugetlb:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
10:net_prio:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
9:perf_event:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
8:net_cls:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
7:freezer:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
6:devices:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
5:memory:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
4:blkio:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
3:cpuacct:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
2:cpu:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
1:cpuset:/docker/795c2b3af01d8186ff10735a1cc8aec33856b85820a50cbd29539e69d3236fe4
bash-5.0# echo $HOSTNAME
795c2b3af01d
```

Reply:
Just found some discussions like
https://forums.docker.com/t/get-a-containers-full-id-from-inside-of-itself/37237/4